browser/components/smartwindow/content/smartwindow.mjs

```diff
@@ -892,43 +863,17 @@ class SmartWindowPage {
       }
     });
 
-    // Setup model picker
+    this.submitButton.addEventListener("selection-change", (e) => {
+      const { value: forcedQueryType } = e.detail;
+      this.effectiveQueryType = forcedQueryType;
+    });
+
+    // Setup model picker (keep existing code)
     this.modelPicker = document.getElementById("model-picker");
     if (this.modelPicker) {
-      // Set initial value from pref
-      this.modelPicker.value = Services.prefs.getStringPref(
-        "browser.smartwindow.model"
-      );
-
-      // Update pref when model changes
+      this.modelPicker.value = Services.prefs.getStringPref("browser.smartwindow.model");
       this.modelPicker.addEventListener("change", () => {
-        Services.prefs.setStringPref(
-          "browser.smartwindow.model",
-          this.modelPicker.value
-        );
-      });
-    }
-
-    // Setup query type picker
-    this.queryTypePicker = document.getElementById("query-type-picker");
-    if (this.queryTypePicker) {
-      // Set initial value from pref (default to "auto")
-      this.queryTypePicker.value = Services.prefs.getStringPref(
-        "browser.smartwindow.queryType",
-        "auto"
-      );
-
-      // Update pref when query type changes
-      this.queryTypePicker.addEventListener("change", () => {
-        Services.prefs.setStringPref(
-          "browser.smartwindow.queryType",
-          this.queryTypePicker.value
-        );
```

Comment:
With the new behavior we don’t need to store the query type override in prefs anymore.